server/src/domain/system-config/system-config.service.ts

```diff
@@ -17,11 +19,24 @@ import { SystemConfigCore, SystemConfigValidator } from './system-config.core';
 @Injectable()
 export class SystemConfigService {
   private core: SystemConfigCore;
+  public availableVersion: ServerVersion | null;
+  private proxyHost: string | null;
+  private proxyProtocol: string | null;
+  private proxyPort: number | null;
+  private disableCheckLatestVersion: boolean;
+
+  private logger = new Logger();
+
   constructor(
     @Inject(ISystemConfigRepository) repository: ISystemConfigRepository,
     @Inject(IJobRepository) private jobRepository: IJobRepository,
   ) {
+    this.availableVersion = null;
     this.core = new SystemConfigCore(repository);
+    this.proxyHost = process.env.PROXY_HOST || null;
+    this.proxyProtocol = process.env.PROXY_PROTOCOL || null;
+    this.proxyPort = Number(process.env.PROXY_PORT) || null;
+    this.disableCheckLatestVersion = process.env.CHECK_NEW_VERSION_INTERVAL === '0';
```

Comment:
Can we make this configurable via the UI instead?

Reply:
Yes, it would be better !